confluence/server/scio_search/src/main/java/com/askscio/atlassian_plugins/confluence/impl/ScioSearchInfoResponse.java

```diff
@@ -0,0 +1,130 @@
+    private final String email;
+    private final boolean isAdmin;
+
+    public UserInfo(UserManager userManager) {
+      final UserProfile profile = userManager.getRemoteUser();
+      this.userKey = profile.getUserKey().getStringValue();
+      this.userName = profile.getUsername();
+      this.fullName = profile.getFullName();
+      this.email = profile.getEmail();
+      this.isAdmin = userManager.isSystemAdmin(profile.getUserKey());
+    }
+  }
+
+  @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+  public static class InstanceInfo {
+
+    private final String version;
+    private final String baseUrl;
+    private final List<InstalledPluginInfo> installedPluginInfos;
+
+    public InstanceInfo(SettingsManager settingsManager, PluginAccessor pluginAccessor,
+        boolean getInstalledPlugins) {
+      this.version = GeneralUtil.getVersionNumber();
+      this.baseUrl = settingsManager.getGlobalSettings().getBaseUrl();
+      if (getInstalledPlugins) {
+        this.installedPluginInfos = new ArrayList<>();
+        Collection<Plugin> installedPlugins = pluginAccessor.getPlugins();
+        for (Plugin plugin : installedPlugins) {
+          InstalledPluginInfo installedPluginInfo = new InstalledPluginInfo(pluginAccessor, plugin);
+          this.installedPluginInfos.add(installedPluginInfo);
+        }
+      } else {
+        this.installedPluginInfos = null;
+      }
+    }
+
+    @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+    public static class InstalledPluginInfo {
+
+      private final String name;
+      private final String key;
+      private final String description;
+      private final String vendorName;
+      private final String vendorUrl;
+      private final String version;
+      private final boolean isEnabled;
+
+      public InstalledPluginInfo(PluginAccessor accessor, Plugin plugin) {
+        PluginInformation pluginInformation = plugin.getPluginInformation();
+        this.name = plugin.getName();
+        this.key = plugin.getKey();
+        this.description = pluginInformation.getDescription();
+        this.vendorName = pluginInformation.getVendorName();
+        this.vendorUrl = pluginInformation.getVendorUrl();
+        this.version = pluginInformation.getVersion();
+        this.isEnabled = accessor.isPluginEnabled(plugin.getKey());
+      }
+    }
+  }
+
+  @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+  public static class PluginInfo {
+
+    private final String version;
+    private final String target;
+    private final String lastWebhookResponseTime;
+    private final String lastWebhookResponseCode;
+    private final String lastWebhookSuccessTime;
+    private final String lastWebhookFailureTime;
+
+    public PluginInfo(PluginAccessor pluginAccessor, PluginSettingsFactory pluginSettingsFactory) {
+      final PluginSettings pluginSettings = pluginSettingsFactory.createGlobalSettings();
+      this.version = pluginAccessor.getPlugin(Constants.PLUGIN_KEY).getPluginInformation()
```

Comment:
do we need a null check at any level in this chain of calls ?

Reply:
i don't think it is required, but anyways added for safety